src/test/regress/expected/bfv_planner.out

```diff
@@ -353,6 +353,184 @@ PL/pgSQL function "volatilefunc" line 4 at assignment
+(10 rows)
+
+explain select * from t_hashdist cross join (select random() as k, sum(a) from generate_series(1, 10) a group by k) x;
+                                               QUERY PLAN                                               
+--------------------------------------------------------------------------------------------------------
+ Gather Motion 3:1  (slice2; segments: 3)  (cost=88.00..4674967.00 rows=77900000 width=28)
+   ->  Nested Loop  (cost=88.00..4674967.00 rows=25966667 width=28)
+         ->  Seq Scan on t_hashdist  (cost=0.00..879.00 rows=25967 width=12)
+         ->  Materialize  (cost=88.00..118.00 rows=1000 width=16)
+               ->  Broadcast Motion 1:3  (slice1)  (cost=20.00..85.00 rows=3000 width=16)
+                     ->  HashAggregate  (cost=20.00..35.00 rows=1000 width=16)
+                           Group By: random()
+                           ->  Function Scan on generate_series a  (cost=0.00..15.00 rows=1000 width=4)
+ Settings:  enable_bitmapscan=off; enable_seqscan=off; optimizer=off
+ Optimizer status: legacy query optimizer
+(10 rows)
+
+explain select * from t_hashdist cross join (select a, count(1) as s from generate_series(1, 10) a group by a having count(1) > random() order by a) x ;
+                                                  QUERY PLAN                                                  
+--------------------------------------------------------------------------------------------------------------
+ Gather Motion 3:1  (slice2; segments: 3)  (cost=147.83..4675026.83 rows=77900000 width=24)
+   ->  Nested Loop  (cost=147.83..4675026.83 rows=25966667 width=24)
+         ->  Seq Scan on t_hashdist  (cost=0.00..879.00 rows=25967 width=12)
+         ->  Materialize  (cost=147.83..177.83 rows=1000 width=12)
+               ->  Broadcast Motion 1:3  (slice1)  (cost=62.33..144.83 rows=3000 width=12)
+                     ->  GroupAggregate  (cost=62.33..94.83 rows=1000 width=12)
+                           Filter: count(1)::double precision > random()
+                           Group By: a.a
+                           ->  Sort  (cost=62.33..64.83 rows=1000 width=4)
+                                 Sort Key: a.a
+                                 ->  Function Scan on generate_series a  (cost=0.00..12.50 rows=1000 width=4)
+ Settings:  enable_bitmapscan=off; enable_seqscan=off; optimizer=off
+ Optimizer status: legacy query optimizer
+(13 rows)
+
+-- limit
+explain select * from t_hashdist cross join (select * from generate_series(1, 10) limit 1) x;
+                                           QUERY PLAN                                           
+------------------------------------------------------------------------------------------------
+ Gather Motion 3:1  (slice2; segments: 3)  (cost=0.00..4974.06 rows=77900 width=16)
+   ->  Nested Loop  (cost=0.00..4974.06 rows=25967 width=16)
+         ->  Broadcast Motion 1:3  (slice1)  (cost=0.00..0.06 rows=3 width=4)
```

Comment:
We don't have a materialize node above `Nested Loop` as we have in 6X. But I don't think it affects the idea of this test. Am I right?

Reply:
Yes